src/endpoint/endpoint.js

```diff
@@ -181,14 +181,33 @@ async function main(options = {}) {
                 should_monitor: nsr => Boolean(nsr.nsfs_config),
             }));
         }
+
+        const report_bulk_sizes = semaphore_report_bulk_sizes();
```

Comment:
@dannyzaken  method will execute only once inside endpoint.js and avoid executing multiple times in BG worker, Since its load values only once when the endpoint pod initializes user has to restart the endpoint pod after updating `SEMAPHORE_METRICS_AVERAGE_INTERVALS` config. Should I move this method to BG worker?

Reply:
My comment was more on the location of the `semaphore_report_bulk_sizes` function. This code is more relevant inside `semaphore_monitor.js` than `endpoint.js`. You can move it to the constructor of `SemaphoreMonitor` and it will only be called once.
It's good to still accept `report_sample_sizes` as one of the constructor parameters so that you can override the config.js defaults  (e.g., for testing purposes).